### ISSUE TITLE
[Issue][Ubuntu 18.04.x LTS] Right before trigger...

### ISSUE THREAD

**Issue by zurabi97** [checkra1n/BugTracker#780]
**Tell us about your setup:**
1. What iDevice are you using? - iPad Mini 3 A1599
2. On what version of iOS is it? -12.4.5
3. What version of checkra1n are you using? GUI or CLI? - Both CLI and GUI.
4. What is your host system (OS version? Hackintosh? VM? etc.)? - PC running Ubuntu.
5. How are you connecting to the device (USB-A? USB-C? Apple/3rd party cable? Through a USB hub?)? - USB-A to lightning.

**What are the steps to reproduce the issue?**
1. 
   1a. Running checkra1n app.
   1b. Following instructions to boot into DFU.
   1c. Stuck at "Right Before trigger..."
2. 
   2a. Running Checkra1n CLI.
   2b. Same as 1b
   2c. Same as 1c

**What do you expect, and what is happening instead?**
Expecting that the device would get jailbroken, but it stuck on "Right before trigger..."

**Does the issue also occur if you enable Safe Mode?**
Didn't tested.

**Any other info, error logs, screenshots, ...?**
![Screenshot from 2020-02-05 15-22-47](https://user-images.githubusercontent.com/29329067/73853141-0e642000-4839-11ea-8faa-8254e9506e36.png)

**Comment by marleyqp** [comment]
Also having this problem.
iPhone 7 Plus, iOS 13.3, Ubuntu 18.04LTS

**Comment by utqwashere** [comment]
Same problem. iPhone 7 iOS 13.3 Ubuntu 18.04.3LTS

**Comment by ghost** [comment]
me too, iphone 7 ios 13.3 ubuntu 19.10. tested safe mode too!

**Comment by marleyqp** [comment]
I ended up download checkra1n through the repo with the GUI and it worked first try. Try doing that.

**Comment by Cattepie** [comment]
> I ended up download checkra1n through the repo with the GUI and it worked first try. Try doing that.

how did you do? commands seems don't work.. 
-echo "deb https://assets.checkra.in/debian / -- this one gives a link
-sudo tee -a /etc/apt/sources.list -- nothing happens and the terminal freezes
-sudo apt-key adv --fetch-keys https://assets.checkra.in/debian/archive.key -- seems working
-sudo apt update -- gives error
-Install the checkra1n package -- actually i don't know how to do it

i'm using the code version

**Comment by utqwashere** [comment]
> I ended up download checkra1n through the repo with the GUI and it worked first try. Try doing that.

I wrote the commands before trying jailbreak but i’m having this situation.

**Comment by zurabi97** [comment]
> I ended up download checkra1n through the repo with the GUI and it worked first try. Try doing that.

I downloaded through the Checkra1n repo.

and I have to force reboot my device to boot it (as jailbreak was usuccessful)

**Comment by tampa-bay** [comment]
Seeing similar as well, with the GUI and CLI on 18.04

Noticed this in the checkra1n log:  
<Verbose>: Deterministic approach doesn't seem to work, attempting to do a probabilistic run

With a lot of this in the kernel log:
usb 1-8: usbfs: USBDEVFS_CONTROL failed cmd checkra1n rqt 128 rq 6 len 64 ret -110

Hope these can provide insight.

**Comment by zurabi97** [comment]
From Ubuntu built-in logcat app:

![Screenshot from 2020-02-05 18-33-53](https://user-images.githubusercontent.com/29329067/73861773-2f7f3d80-4846-11ea-86e5-69069de04a20.png)

**Comment by ghost** [comment]
> I ended up download checkra1n through the repo with the GUI and it worked first try. Try doing that.

true, it worked, but downloading it crashed and i have to restore my phone.. stuck in dfu mode..

**Comment by zurabi97** [comment]
> > I ended up download checkra1n through the repo with the GUI and it worked first try. Try doing that.
> 
> true, it worked, but downloading it crashed and i have to restore my phone.. stuck in dfu mode..

You reinstalled the checkra1n?

**Comment by zurabi97** [comment]
Reinstalling checkra1n does not solve the issue.

**Comment by zurabi97** [comment]
UPD: I enabled through options the "Allow untested iOS/iPadOS/tvOS versions" and sometimes get error "DFUSyncUpload FAILED: -1"

**Comment by utqwashere** [comment]
I tried on linux mint and it worked.

**Comment by marleyqp** [comment]
Is working now and I’m using checkra1n downloaded through the repo / using GUI

**Comment by zurabi97** [comment]
> I tried on linux mint and it worked.

Live usb?

**Comment by utqwashere** [comment]
> > I tried on linux mint and it worked.
> 
> Live usb?

idk i just installed linux mint 19.3 mate to my pc and tried that guy’s method it works: https://youtu.be/YtIVyotMx1Y

**Comment by zurabi97** [comment]
UPD: Tried with Linux Mint 19.3 (Mate DE) live usb and has the same thing.

**Comment by marleyqp** [comment]
It leads me to believe that whatever OS you try, it won’t work. Have you tried bootra1n?

**Comment by zurabi97** [comment]
Tried with Bootra1n still the same.

**Comment by zurabi97** [comment]
Basically, in logs I get error that checkra1n is being blocked for some reason.

**Comment by marleyqp** [comment]
What actually happens when you attempt to jailbreak?

**Comment by zurabi97** [comment]
Well, happens this:
[] INFO: task checkra1n:xxxx blocked for more than X seconds.
[] Not tainted 5.4.17_1 [hashtag]1
[] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.

**Comment by truff1ep1g** [comment]
I have tried this method in mint and Ubuntu. Through cli and gui both. All combinations of checkra1n options(-v on and off, safe on and off, untested on and off,etc....) I either get these results. I either get stuck at entered download mode at which my phone shows some commands the checkra1n logo and then shows the apple logo and reboots to normal mode with no checkra1n logo on the screen. Or I run checkra1n and my phone gets stuck in dfu mode and I have to manually restart it and the checkra1n GUI wil be stuck on entered download mode.  If I run through cli I get error 21 or erro 20. I have switched ports on my laptop. Made sure I’m using the 3.0. The one marked with “ss”. Tried 4 different Apple branded cables. Even a brand new one that came with airpods tried off brand cables as well. anything I have read that is said to have fixed these issues. I’ve deleted checkra1n through terminal installed through the repo.  Applied usbmuxd commands. And I still get no logo on my home screen.

**Comment by truff1ep1g** [comment]
Try this it works. I’ve been stuck too. looked for days then ran across this. Forget all those YouTube videos. This did iPhone X,13 .3, iPhone 7, 13.3, iPad air2 , 13.3, and iPhone 8+ ,13.3. All in first try with different cables on different ports.

https://github.com/foxlet/bootra1n/blob/master/README.md

**Comment by zurabi97** [comment]
UPD: Tried on another laptop with bootra1n and not worked. It seems that the app have bug when trying to jailbreak an A7 device.

**Comment by 141** [comment]
Tried bootrain 10.1 using two different cables and two different USB ports on one computer so far and I still get the 122 second timeout for a 5S on 12.4.4

**Comment by zurabi97** [comment]
> Tried bootrain 10.1 using two different cables and two different USB ports on one computer so far and I still get the 122 second timeout for a 5S on 12.4.4 

Try #1169

**Comment by 141** [comment]
I did and it worked, even though the directions were way more complicated then they need be. At 122 second timeout I unplugged and quickly replugged the lightning port and saw the jailbreak happen quickly on the phone screen on my first try doing that.

**Comment by ujaveri** [comment]
iphone 8  ios: 12.4.1
system: Sony Vaio ( LVGN-CR320E 4.15.0-97-generic #98-Ubuntu SMP Tue Apr 7 10:29:57 UTC 2020 i686 i686 i686 GNU/Linux ) Ubuntu 18.04.4 LTS
cli version: checkra1n-i486

same error : Right before trigger

**Comment by ujaveri** [comment]
Same error with Bootra1n latest checkra1n build
On ibm Thinkpadz
Stuck at Right before trigger....

**Comment by abuet2004** [comment]
I have also been stuck at right before trigger with both "122 seconds" error and script displaying INFO: task checkra1n:xxxx blocked for more than X seconds.
[] Not tainted 5.4.17_1 [hashtag]1
[] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message. The script repeats itself when uplug and plug back in at right before trigger. I have enabled all settings and im using an ipad air 2 on ios 12.5.5
